File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/ObjectElementDescriptor.java
package org.opcfoundation.webserver.types;

import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.eclipse.milo.opcua.stack.core.types.builtin.NodeId;

import org.opcfoundation.webserver.digitaltwin.element.ElementType;

public class ObjectElementDescriptor {
    private final String id;
    private final LocalizedText displayName;
    private final NodeId typeId;

    public ObjectElementDescriptor(
            String elementId,
            LocalizedText displayName,
            ElementType elementType)
    {
        this.id = elementId;
        this.displayName = displayName;
        this.typeId = elementType.nodeId();
    }

    public String getId() {
        return id;
    }

    public LocalizedText getDisplayName() {
        return displayName;
    }

    public NodeId getTypeId() {
        return typeId;
    }
}

File: opcua-webservice-node/src/server/types/digitaltwin/ObjectElementDescriptor.ts
import { UaLocalizedText, UaNodeId } from "opcua-webapi-ts";
import { UaObjectType } from "../../addressspace/nodes/UaObjectType";

export class ObjectElementDescriptor {
    private readonly _id: string;
    private readonly _displayName: UaLocalizedText;
    private readonly _typeId: UaNodeId;

    constructor(elementId: string, displayName: UaLocalizedText, elementType: UaObjectType) {
        this._id = elementId;
        this._displayName = displayName;
        this._typeId = elementType.nodeId;
    }

    get id(): string {
        return this._id;
    }

    get displayName(): UaLocalizedText {
        return this._displayName;
    }

    get typeId(): UaNodeId {
        return this._typeId;
    }
}